### PR TITLE
Add auto-prefixer to compilation step

### PR DESCRIPTION
**Pull request by sndrs**
Auto-prefix the CSS, with a view to being able to remove mixins, automatically support/drop support and easily handle cases like `transition: transform()` (e.g. in #8834).

Once this is in, we can start to remove old CSS3 mixins, one by one.

Note that this _does_ change our current CSS (mostly removing rules that are no longer required) so it's not a simple drop in.

There's zip of all the diffs here: http://cl.ly/3y3z263w0M1n, but here's some examples:
- global: http://www.diffnow.com/?report=84w5j
- IE9.global: http://www.diffnow.com/?report=gqx5c
### Testing

This will need pretty thorough checking I guess @michaelwmcnamara, hard to say what will be affected, but fronts especially.

UPDATE - checking the diffs, the following areas worth double checking:
- social icons
- most popular
- show more loading animation 
- video player controls
- soulmates
- fronts

Anything you think it's important to check, but I'd say these specifically?
- [x] IE8
- [x] IE9
- [x] IE10
- [x] IE11
- [x] Chrome 36
- [x] Chrome latest
- [x] Firefox 26
- [x] Firefox latest
- [x] Safari 5
- [x] Safari 6
- [x] Safari 7
- [x] Safari 8
- [ ] iOS 5
- [x] iOS 6
- [x] iOS 7
- [x] iOS 8
- [x] Android 2s
- [x] Android 4s
- [ ] Windows phone 7
- [x] Windows phone 8
  old crappy devices:
- [x] iPad2
- [x] iphone4S

@phamann anything else you think this should do? @davidrapson does this look reasonable to you?

### CODE REVIEW

File: grunt-configs/autoprefixer.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+        },
+        ie9: {
+            files: [{
+                expand: true,
+                cwd: dir,
+                src: ['ie9.*.css'],
+                dest: dir
+            }],
+            options: {
+                browsers: ['ie 9']
```
Comment (davidrapson) [inline grunt-configs/autoprefixer.js]: `Explorer 8` here or `ie` everywhere else for consistency?

File: grunt-configs/autoprefixer.js [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+        },
+        'old-ie': {
+            files: [{
+                expand: true,
+                cwd: dir,
+                src: ['old-ie.*.css'],
+                dest: dir
+            }],
+            options: {
+                browsers: ['Explorer 8']
```
Comment (davidrapson) [inline grunt-configs/autoprefixer.js]: Just IE8?

**Comment by michaelwmcnamara** [comment]
image

File: grunt-configs/autoprefixer.js (hunk 1 above)
Comment (sndrs) [inline grunt-configs/autoprefixer.js]: Yeah good point

**Comment by davidrapson** [comment]
Looks reasonable to me.

Most vendor prefixes are pretty safe to drop as *so* much can be un-prefixed now and a chunk of typically prefixed properties are visual enhancements anyway.

The major exception to this is `flexbox`, which is pretty critical to layout in your case. Thats the one area where you might need to tweak the config to make sure you've covered all the different flexbox syntaxes…that said your `browserlist` file looks like it should cover it OK. Better to be conservative with those settings anyway.

File: grunt-configs/autoprefixer.js (hunk 2 above)
Comment (sndrs) [inline grunt-configs/autoprefixer.js]: yeah i think so - is there anything in IE7 that needs prefixing?

Comment (davidrapson) [inline grunt-configs/autoprefixer.js]: Nope, I didn't think there was anything in IE8 that did either though.

Comment (davidrapson) [inline grunt-configs/autoprefixer.js]: Think my point is that I'm not sure what the setting needs to be here for a stylesheet that's just served to IE8.

**Comment by robertberry** [comment]
Nice! :+1: How many Kbs do we save when we remove prefixes?

File: grunt-configs/autoprefixer.js (hunk 2 above)
Comment (sndrs) [inline grunt-configs/autoprefixer.js]: ah yeah, fair enough. currently, we\re still using the mixins (keep this pr smaller) so it's actually removing prefixes here.

**Comment by sndrs** [comment]
@michaelwmcnamara updated the description to include some areas that have notable changes in the diffs

**Comment by michaelwmcnamara** [comment]
@sndrs Thanks Alex, I'll make sure those areas get some special attention.

File: browserslist [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+
+Firefox >= 26
+Explorer >= 10
+Safari >= 5
+Chrome >= 36
+
+iOS >= 5
+Android >= 2
+BlackBerry >= 6
+ExplorerMobile >= 7
```
Comment (phamann) [inline browserslist]: 7??

**Comment by phamann** [comment]
How are we going to ensure that we keep the [canisue database](https://github.com/nDmitry/grunt-autoprefixer#updating-prefixes-database) up-to-date?

**Comment by phamann** [comment]
Have you tested the source map output?

**Comment by phamann** [comment]
As @robertberry mentions, I'm very interested in the file size differences. i'm assuming this output should be smaller as we are removing unused prefixes?

**Comment by phamann** [comment]
:+1: Other than comments

File: browserslist (hunk 3 above)
Comment (sndrs) [inline browserslist]: win7 is not even level 3 support, think this should be higher or lower?

**Comment by sndrs** [comment]
> How are we going to ensure that we keep the canisue database up-to-date?

got spot, will add that step to compile

**Comment by sndrs** [comment]
> How many Kbs do we save when we remove prefixes?

not massive savings, esp when gzipped. the benefit is really more accurate attributes, cleaner sass and support for complex prefixed rules...

File | Current | Auto-prefixed
------------ | ------------- | -------------
facia.css | 158K  | 157K
facia.css.gz | 22K  | 22K
global.css | 460K  | 455K
global.css.gz | 73K  |  73K
head.content.css | 147K  | 147K
head.content.css.gz | 20K  |  20K
head.facia.css | 173K  | 172K
head.facia.css.gz | 22K   | 22K
head.football.css | 104K | 105K
head.football.css.gz | 17K |  17K
head.footballSnaps.css | 25K   | 25K
head.footballSnaps.css.gz | 3.7K  | 3.7K
head.identity.css | 87K  |  87K
head.identity.css.gz | 14K  |  14K
head.index.css | 70K |   70K
head.index.css.gz | 11K  |  11K
head.rich-links.css | 202K  | 201K
head.rich-links.css.gz | 36K |   35K
ie9.global.css | 460K |  436K
ie9.global.css.gz | 73K   | 71K
ie9.head.content.css | 147K |  143K
ie9.head.content.css.gz | 20K   | 20K
ie9.head.facia.css | 173K |  162K
ie9.head.facia.css.gz | 22K  |  21K
ie9.head.football.css | 104K  | 100K
ie9.head.football.css.gz | 17K  |  16K
ie9.head.identity.css | 87K  |  84K
ie9.head.identity.css.gz | 14K |   14K
ie9.head.index.css | 70K  |  67K
ie9.head.index.css.gz | 11K  |  11K
media-player.css | 134K |  134K
media-player.css.gz | 28K   | 29K
membership-icons.css | 14K  |  14K
membership-icons.css.gz | 4.4K |  4.4K
old-ie.global.css | 282K  | 263K
old-ie.global.css.gz | 41K  |  40K
old-ie.head.content.css | 107K  | 104K
old-ie.head.content.css.gz | 17K |   16K
old-ie.head.facia.css | 126K  | 116K
old-ie.head.facia.css.gz | 18K  |  17K
old-ie.head.football.css | 73K  |  70K
old-ie.head.football.css.gz | 14K  |  13K
old-ie.head.identity.css | 64K  |  60K
old-ie.head.identity.css.gz | 12K  |  12K
old-ie.head.index.css | 50K   | 47K
old-ie.head.index.css.gz | 9.6K  | 9.3K
print.css | 5.4K  | 5.4K
print.css.gz | 2.6K  | 2.6K
webfonts-hinting-auto-kerning-on.css | 11K  |  11K
webfonts-hinting-auto-kerning-on.css.gz | 706B  | 706B
webfonts-hinting-cleartype-kerning-on.css | 11K |   11K
webfonts-hinting-cleartype-kerning-on.css.gz | 714B  | 714B
webfonts-hinting-off-kerning-on.css | 11K   | 11K
webfonts-hinting-off-kerning-on.css.gz | 704B |  704B

**Comment by sndrs** [comment]
>Have you tested the source map output? 

yeah they seem fine

**Comment by phamann** [comment]
Whats the status of this?

**Comment by sndrs** [comment]
in testing, they're on it

**Comment by sndrs** [comment]
- [ ] testing 123

**Comment by ScottPainterGNM** [comment]
Bugs Summary

- [ ] ipad1 & 2: Network fronts should only be serving a shortened front for iPad 1 and 2 (or iOS5 and 6) which shows “Around the World” as the final container and does not show in brief, in depth, take part, pictures and video, people and most popular. This is currently not showing all of these on the ipad.

- [ ] ipad air: mpu in most popular container is not loading - load if focus is on container at time of refresh but not if you load page with focus on top nav then scroll down to most popular container - possible issue with lazy-loading?

- [ ] Nexus 7 (android 4.4.4): Soulmates 404 not found error: (further investigation: soulmates container at bottom of tag page loads ok on ipad air - there could be something wrong with its loading on curated fronts - worth a look)


@michaelwmcnamara may have more as he is finishing up on the Windows 7 box but will update this list if he finds anything

**Comment by sndrs** [comment]
>  ipad air: mpu in most popular container is not loading 

cannot recreate, putting down to a dodgy add

**Comment by michaelwmcnamara** [comment]
yep - I do!
- [ ] Firefox latest on Windows 7: Video playables on fronts are not playing.
 See https://docs.google.com/a/guardian.co.uk/document/d/19QlOyQVdFyxors8mhV1hyrX_zYnpomDQbGIkfJZgQZ8/edit?usp=sharing 
for screenshot with console debugger messages

**Comment by sndrs** [comment]
> Windows 7 - Firefox latest (with latest flash) On audio pages, popular in podcast container does not show images

this is a bug on prod too - seems like working images are swapped out for something non-working. def worth raising but not caused by this PR

**Comment by sndrs** [comment]
> Firefox 26/Safari 6 - Soulmates missing (found on boxing page in prod/other browsers)

also on prod. happy for this PR to ignore

**Pull request merged**